convert2rhel/subscription.py

```diff
@@ -127,50 +127,40 @@ def get_registration_cmd():
     """Build a command for subscription-manager for registering the system."""
     loggerinst.info("Building subscription-manager command ... ")
     registration_cmd = "subscription-manager register --force"
+
+    loggerinst.info("Checking for activation key ...")
     if tool_opts.activation_key:
         # Activation key has been passed
         # -> username/password not required
         # -> organization required
         loggerinst.info("    ... activation key detected: %s" % tool_opts.activation_key)
-        registration_cmd += ' --activationkey="%s"' % tool_opts.activation_key
+
+        # TODO: Parse the output of 'subscription-manager orgs' and let the
+        # user choose from the available organizations. If there's just one,
+        # pick it automatically.
+        # Organization is required when activation key is used
+        org = tool_opts.org
+        while not org:
+            org = utils.prompt_user("Organization: ")
+
+        registration_cmd += ' --activationkey="%s" --org=%s' % (tool_opts.activation_key, org)
     else:
-        # No activation key -> username/password required
-        if tool_opts.username and tool_opts.password:
-            loggerinst.info("    ... activation key not found, using given username and password")
-        else:
-            loggerinst.info("    ... activation key not found, username and password required")
+        loggerinst.info("    ... activation key not found, username and password required")
 
-        if tool_opts.username:
-            loggerinst.info("    ... username detected")
-            username = tool_opts.username
-        else:
+        username = tool_opts.username
+        while not username:
             username = utils.prompt_user("Username: ")
-        if tool_opts.password:
-            loggerinst.info("    ... password detected")
-            password = tool_opts.password
-        else:
-            if tool_opts.username:
-                # Hint user for which username they need to enter pswd
-                loggerinst.info("Username: " + username)
```

Comment:
It would be preferential to keep the logs here whenever we detect that a value exists in toolopts. Likewise a log should be added if we detect an org in toolopts

The reason being that we're being explicit in the logic of what's going on. Rather than just logs in the beginning that we found specific values in the parsing of the CLI args.